### PR TITLE
Propogate checkUsageLeak options

### PR DESCRIPTION
**Pull request by pranjalssh**

### CODE REVIEW

**Review by xiaoxmeng — COMMENTED** [review]
@pranjalssh thanks for the change. LGTM. Could you add a test that glag is true but the options passed to the memory manager is false. And verify the check leak is false from the memory pool allocated from the memory manager? We can provide testingcheckUsageLeak() in memory pool? Thanks!

**Comment by pranjalssh** [comment]
@xiaoxmeng updated

File: velox/common/memory/tests/MemoryManagerTest.cpp [hunk 1]
```diff
@@ -483,6 +483,17 @@ TEST(MemoryManagerTest, quotaEnforcement) {
...
 
+TEST(MemoryManagerTest, testCheckUsageLeak) {
+  FLAGS_velox_memory_leak_check_enabled = true;
+  auto& manager = MemoryManager::getInstance(
+      memory::MemoryManager::Options{.checkUsageLeak = false}, true);
+
+  auto rootPool = manager.addRootPool("duplicateRootPool", kMaxMemory);
+  auto leafPool = manager.addLeafPool("duplicateLeafPool", true);
+  ASSERT_EQ(rootPool->testingCheckUsageLeak(), false);
+  ASSERT_EQ(leafPool->testingCheckUsageLeak(), false);
```
Comment (xiaoxmeng) [inline velox/common/memory/tests/MemoryManagerTest.cpp:494]: Could you also check the memory pool created through MemoryManager::addLeafPool? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@pranjalssh LGTM. Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@pranjalssh LGTM. Thanks!

File: velox/common/memory/tests/MemoryManagerTest.cpp (hunk 1 above)
Comment (pranjalssh) [reply velox/common/memory/tests/MemoryManagerTest.cpp:494]: I don't understand, this is exactly what this line checks

**Pull request closed without merging**